network-store-server/src/main/java/com/powsybl/network/store/server/NetworkStoreRepository.java

```diff
@@ -2095,6 +2217,52 @@ private Map<OwnerInfo, List<ReactiveCapabilityCurvePointAttributes>> innerGetRea
         }
     }
 
+    protected <T extends AbstractIdentifiableAttributes> Map<OwnerInfo, RegulationPointAttributes> getRegulationPointFromEquipment(UUID networkUuid, List<Resource<T>> resources) {
+        Map<OwnerInfo, RegulationPointAttributes> map = new HashMap<>();
+        if (!resources.isEmpty()) {
+            for (Resource<T> resource : resources) {
+                OwnerInfo info = new OwnerInfo(
+                    resource.getId(),
+                    resource.getType(),
+                    networkUuid,
+                    resource.getVariantNum()
+                );
+                map.put(info, resource.getAttributes().getRegulationPoint());
+            }
+        }
+        return map;
+    }
+
+    private Map<OwnerInfo, RegulationPointAttributes> innerGetRegulationPoints(PreparedStatement preparedStmt, ResourceType type) throws SQLException {
+        try (ResultSet resultSet = preparedStmt.executeQuery()) {
+            Map<OwnerInfo, RegulationPointAttributes> map = new HashMap<>();
+            while (resultSet.next()) {
+                OwnerInfo owner = new OwnerInfo();
+                RegulationPointAttributes regulationPointAttributes = new RegulationPointAttributes();
+                // In order, from the QueryCatalog.buildRegulationPointQuery SQL query :
+                // equipmentId, networkUuid, variantNum, regulatedEquipmentId, localTerminal and regulatingTerminal
+                String regulatedEquipmentId = resultSet.getString(3);
+                owner.setEquipmentId(regulatedEquipmentId);
+                owner.setNetworkUuid(UUID.fromString(resultSet.getString(1)));
+                owner.setVariantNum(resultSet.getInt(2));
+                owner.setEquipmentType(type);
+                regulationPointAttributes.setRegulatedEquipmentId(regulatedEquipmentId);
+                regulationPointAttributes.setRegulationMode(resultSet.getString(4));
+                regulationPointAttributes.setIdentifiableType(IdentifiableType.valueOf(type.toString()));
+                Optional<String> localConnectableId = Optional.ofNullable(resultSet.getString(5));
+                if (localConnectableId.isPresent()) {
+                    regulationPointAttributes.setLocalTerminal(new TerminalRefAttributes(localConnectableId.get(), resultSet.getString(6)));
+                }
+                Optional<String> regulatingConnectableId = Optional.ofNullable(resultSet.getString(7));
+                if (regulatingConnectableId.isPresent()) {
+                    regulationPointAttributes.setRegulatingTerminal(new TerminalRefAttributes(resultSet.getString(7), resultSet.getString(8)));
```

Comment:
regulationPointAttributes.setRegulatingTerminal(new TerminalRefAttributes(regulatingConnectableId.get(), resultSet.getString(8)));

Reply:
regulatingConnectableId is the the id of the equipment that regulate. well i think the name is not good

Reply:
but we can wirte 
regulationPointAttributes.setLocalTerminal(new TerminalRefAttributes(regulatingConnectableId.get(), resultSet.getString(8)));